apps/internal/shared/shared.go

```diff
@@ -47,5 +47,38 @@ func (acc Account) Key() string {
 	return strings.Join([]string{acc.HomeAccountID, acc.Environment, acc.Realm}, CacheKeySeparator)
 }
 
+//IsZero checks the zero value of account
+func (acc Account) IsZero() bool {
```

Comment:
This is getting long in the tooth.  I'm going to suggest using reflection here.

Here is the example in the playground: https://play.golang.org/p/AyFv-rmIjG4
```go
func (a Account) IsZero() bool {
	v := reflect.ValueOf(a)
	for i := 0; i < v.NumField(); i++ {
		field := v.Field(i)
		if !field.IsZero() {
			switch field.Kind() {
			case reflect.Map, reflect.Slice:
				if field.Len() == 0 {
					continue
				}
			}
			return false
		}
	}
	return true
}
```

This can be made even shorter, but a zero length AdditionalFields instead of nil will cause it to think it is not zero. 
It depends on how diligent you are on keeping it nil:

```go
func (a Account) IsZero() bool {
	return reflect.ValueOf(a).IsZero()
}
```

Reply:
I saw there were two implementations of IsZero() in the codebase :
[this](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/1a6b20feab85f7e1afc14529ee38962c97b6ee1f/apps/internal/oauth/ops/accesstokens/tokens.go#L83) and [this](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/35b0b9bbf5df09a3154c841144a91822b1b49c4a/apps/internal/base/internal/storage/items.go#L154)
 I wasn’t sure if we should use the reflect thing for IsZero() or a switch case. Had a discussion with @keegan-caruso  that using reflection might get slower but the switch case can cause maintenance issues. But since account object is pretty standard meaning it is unlikely to change, we decided to go with the faster one - switch case. 
 
 What are your thought ? @element-of-surprise 
 
 I have changed it to the one you suggested in the latest commit to see how long the integration tests take to run with this approach and looks like the time it takes is pretty similar to the one with a switch case. So may be using the reflection would probably be better.

Reply:
the difference in speed will be much less than measurable at the integration test level. ms at most what I would predict, but would have to benchmark it to actually know.

This is more simple.

Reply:
My thoughts are the speed is inconsequential.  You could benchmark it but I doubt it is worth it. 

I like the maintenance friendly version and I would probably just generalize this and use it everywhere.

Reply:
👍

Reply:
@element-of-surprise , @keegan-caruso - Created PR #177 to change it in the rest of the code. This already uses the one suggested by John.